### PR TITLE
Add ESLint, base Airbnb config and Appveyor

### PR DESCRIPTION
**Pull request by nkbt**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -9,4 +9,5 @@ machine:
 
 test:
   override:
-    - cd test && REPORT_DIR=${CIRCLE_TEST_REPORTS} LOG_DIR=${CIRCLE_ARTIFACTS} node ../index.js
+    - $(npm bin)/eslint --debug . --format tap | $(npm bin)/tap-xunit > ${CIRCLE_TEST_REPORTS}/lint.xml && test ${PIPESTATUS[0]} -eq 0
```
Comment (LorbusChris) [inline circle.yml:12]: ~~maybe try  `./node_modules/.bin/eslint --debug . --format tap | ./node_modules/.bin/tap-xunit > ${CIRCLE_TEST_REPORTS}/lint.xml && test ${PIPESTATUS[0]} -eq 0`~~

PPS: nevermind, is a thing :smile: and quite nice n short too!

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+branches:
```
Comment (nkbt) [inline appveyor.yml:1]: hopefully will get it running on Appveyor too

**Comment by nkbt** [comment]
Oh wow Appveyor worked out well. Now I would be able to add e2e tests under windows too. That is nice

**Comment by LorbusChris** [comment]
nvmd I just forgot so set the port in the nightwatch CI settings..

**Pull request merged**